docs/technical_reference/costing/automate_detailed_unit_costing_rst_file.py

```diff
@@ -0,0 +1,191 @@
+                    )
+                    f.write(".. csv-table::\n")
+                    f.write(
+                        '   :header: "Description", "Symbol", "Parameter Name", "Default Value", "Units"\n\n'
+                    )
+                    f.write(
+                        '   "description", ":math:`Symbol_{example}`", "parameter_name", "1", ":math:`\\text{dimensionless}`"\n'
+                    )
+
+                    # TODO: add var tables
+                    f.write("\nCosting Method Variables\n")
+                    f.write("+" * len("Costing Method Variables"))
+                    f.write(
+                        f"\n\nThe following variables are constructed when applying the `cost_{unit_name_list[i]}` costing method in the ``watertap_costing_package``:\n\n"
+                    )
+                    f.write(".. csv-table::\n")
+                    f.write(
+                        '   :header: "Description", "Symbol", "Variable Name", "Default Value", "Units"\n\n'
+                    )
+                    f.write(
+                        '   "description", ":math:`Symbol_{example}`", "variable_name", "1", ":math:`\\text{dimensionless}`"\n'
+                    )
+
+                    # TODO: add capex eqs
+                    f.write("\nCapital Cost Calculations\n")
+                    f.write("+" * len("Capital Cost Calculations"))
+                    f.write(
+                        "\n\nDescribe capital costs..keep it concise where possible\n\n"
+                    )
+                    f.write("    .. math::\n\n")
+                    f.write(
+                        "        C_{cap,tot} = C_{cap,example1}+C_{cap,example2}+C_{cap,other}"
+                    )
+                    f.write("\n\n    .. math::\n\n")
+                    f.write(
+                        "        C_{cap,example1} = fill in equation for each component in total capex equation\n\n "
+                    )
+
+                    # TODO: add opex eqs
+                    f.write("\nOperating Cost Calculations\n")
+                    f.write("+" * len("Operating Cost Calculations"))
+                    f.write(
+                        "\n\nDescribe operating/maintenance costs..keep it concise where possible\n\n"
+                    )
+                    f.write("    .. math::\n\n")
+                    f.write(
+                        "        C_{op,tot} = C_{op,example1}+C_{op,example2}+C_{op,other}"
+                    )
+                    f.write("\n\n    .. math::\n\n")
+                    f.write(
+                        "        C_{op,example1} = fill in equation for each component in total opex equation\n\n "
+                    )
+
+                    # TODO: add module directives to unit and cost method
+                    f.write("\nCode Documentation\n")
```

Comment:
I'm assuming the author can should just include both the 0D and 1D models here, or other instances where unit_model:cost_model aren't 1:1. Say the some units having more dependencies on rectifier, heat exchanger, pumps, etc.

Reply:
Because of that, I don't know if it makes sense to have once section as the model _directly_ using the method/class and an optional section for other costing dependencies. Just a thought.

Reply:
So instead of including any link to unit models here, I can just include the cost method, and later, within unit model tech refs, we can point to the relevant default costing method for the unit at hand. For example, in the tech ref for GAC, under costing section, We would just have a sentence pointing to the default costing method used and its tech ref (or something like that).